Mzlib readme (#2391)
dotnet/runtimelab: README.md

# .NET Runtime - Managed Zlib

> This is an experimental managed version, based on madler/zlib and deflate64 project, of ZLib's library, for exploring the idea of owning a competitive compression library that would follow the same cross-platform vision as .NET while having more consistent performance and compression characteristics. The project includes [Deflate algorithm](https://www.zlib.net/feldspar.html)'s implementation for raw inflate, based on [Deflate64](https://github.com/dotnet/runtime/tree/main/src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateManaged), and raw deflate, based on [Zlib](https://github.com/madler/zlib), with a test harness for performance and functionality.

Reviewer: ```suggestion
This is an experimental managed version of ZLib which follows the same cross-platform vision as .NET while having more consistent performance and compression characteristics. The project includes [Deflate algorithm](https://www.zlib.net/feldspar.html)'s implementation for raw inflate, based on [Deflate64](https://github.com/dotnet/runtime/tree/6387a9eb56098a889021190747d31f07246dd9f2/src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateManaged), and raw deflate, based on [Zlib](https://github.com/madler/zlib), with a test harness for performance and functionality.
```

> Consuming an external dependency, like [Zlib](https://github.com/madler/zlib) or [Zlib-intel](https://github.com/intel/zlib), involves high-maintenance cross-platform and .NET versions, being shipped differently for Windows and Linux systems, and a vulnerability when it comes to fix reported issues or bugs. This library is being consumed by compression solutions like [DeflateStream](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.deflatestream?view=net-7.0), [GZipStream](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.gzipstream?view=net-7.0) , [ZLibStream](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.zlibstream?view=net-7.0) and [ZipArchive](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.ziparchive?view=net-7.0), among others.

Reviewer: ```suggestion
Consuming an external dependency like [Zlib](https://github.com/madler/zlib) or [Zlib-intel](https://github.com/intel/zlib) in .NET is a high maintenance task, since it involves shipping different versions for Windows and Unix systems, as well as having to either wait for vulnerabilities to get fixed upstream when they become public, or having to fix them ourselves, causing our code to potentially diverge from the original sources.

This library is being consumed by compression solutions like [DeflateStream](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.deflatestream?view=net-7.0), [GZipStream](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.gzipstream?view=net-7.0) , [ZLibStream](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.zlibstream?view=net-7.0) and [ZipArchive](https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.ziparchive?view=net-7.0), among others.
```
